src/tests/ftest/util/dmg_utils.py

```diff
@@ -628,7 +628,7 @@ def pool_list(self):
         output = self._get_json_result(("pool", "list"))
 
         data = {}
-        if output["response"] is None:
+        if output["response"] is None or output["response"]["pools"] is None:
```

Comment:
This looks fine, but if we wanted to save a bit more space, we could replace lines 631-636 with:
```
        if "response" in output and "pools" in output["response"]:
            for pool in output["response"]["pools"]:
                data[pool["uuid"]] = pool["svc_reps"]
        return data
```